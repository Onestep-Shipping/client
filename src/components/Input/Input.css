.input-group {
  display: flex;
  flex-direction: column;
  margin-top: 5px;
  margin-bottom: 15px;
}

.login-label {
  font-size: 18px;
  font-weight: 600;
  padding-left: 5px;
  text-align: start;
}

.login-input {
  width: 230px;
  border-radius: 3px;
  border: 1px solid #474747;
  transition: border 0.4s ease-out;
  margin: 10px 0px 10px 0px;
  padding: 15px 20px;
  font-family: Oxygen, sans-serif;
  font-size: 16px;
  color: #3e3e42;
  background: rgba(15, 15, 15, 0.01);
  outline: none;
}

.login-input:focus {
  outline: thin;
  box-shadow: 0px 0px 20px rgba(15, 15, 15, 0.2);
}

.login-input::placeholder {
  font-family: Oxygen, sans-serif;
  font-size: 16px;
  color: rgba(15, 15, 15, 0.4);
}

.display-errors:invalid {
  border-width: 1px;
  border-color: red;
}