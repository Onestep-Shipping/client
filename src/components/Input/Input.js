import React from 'react';
import './Input.css';

const Input = (props) => {
  const {type, name, displayErrors} = props;
  const lcName = name.toLowerCase().replace(' ', '-');
  const errorClass = displayErrors ? 'display-errors' : '';

  return (
    <div className="input-group">
      <label htmlFor={lcName} className="login-label">{name}</label>
      <input
        type={type}
        name={lcName}
        className={'login-input ' + errorClass}
        placeholder={name}
        required />
    </div>
  );
};

export default Input;

Input.propTypes = {
  name: String,
  type: String,
  displayErrors: Object,
};
