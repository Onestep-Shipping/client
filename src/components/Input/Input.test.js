import Input from './Input';
import React from 'react';
import { shallow } from 'enzyme';

const label = 'User Email';
let wrapped = shallow(<Input name={label} type="email" displayErrors={false} />);
describe('Input', () => {
  it('should render the Input Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });
  it('should renders the Input label', () => { 
    expect(wrapped.find('label').text()).toEqual(label);
  });
});