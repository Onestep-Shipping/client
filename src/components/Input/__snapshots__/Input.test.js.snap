// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render the Input Component correctly 1`] = `
<div
  className="input-group"
>
  <label
    className="login-label"
    htmlFor="userEmail"
  >
    User Email
  </label>
  <input
    className="login-input "
    name="userEmail"
    placeholder="User Email"
    required={true}
    type="email"
  />
</div>
`;
